openff/toolkit/topology/molecule.py

```diff
@@ -4592,38 +4605,52 @@ def from_mapped_smiles(
     @requires_package("qcelemental")
     def from_qcschema(
         cls: type[FM],
-        qca_record,
-        client=None,
+        qca_object,
         toolkit_registry=GLOBAL_TOOLKIT_REGISTRY,
         allow_undefined_stereo: bool = False,
-    ) -> FM:
+    ):
         """
         Create a Molecule from a QCArchive molecule record or dataset entry
         based on attached cmiles information.
 
-        For a molecule record, a conformer will be set from its geometry.
+        If this method is provided a QCElemental Molecule (or dict representation of a Molecule),
+        it will return a single-conformer OpenFF Molecule.
 
-        For a dataset entry, if a corresponding client instance is provided,
-        the starting geometry for that entry will be used as a conformer.
+        If this method is provided a QCFractal dataset Entry (or dict representation
+        of an Entry), it will return an OpenFF Molecule with at least one conformer, corresponding
+        to the:
+
+        * ``.molecule`` attribute of a single point entry (single conformer)
+        * ``.initial_molecule`` attribute of an optimization or grid optimization entry (single conformer)
+        * ``initial_molecules`` attribute of a torsion drive entry (one or more conformers)
+
+        If these QC molecules have their ``.id`` fields populated, the returned OpenFF Molecule  will have a
+        dict mapping QC IDs to conformer numbers (``offmol.properties["initial_molecules"]``)
+
+        The data source must also specify the kekule structure of the molecule. Currently the only
+        supported format for this is in the ``canonical_isomeric_explicit_hydrogen_mapped_smiles``
+        field, which will be taken from the following locations, if available, in the following
+        order of priority:
+
+        * The input's ``attributes`` attribute (set on Entries)
```

Comment:
Is "Entry" a specific class, a set of classes, or a looser jargon?

Reply:
Entries are a [set of classes](https://github.com/MolSSI/QCFractal/blob/32a29a86d6673b0ea1a743b02d3af314c2fd164c/qcportal/qcportal/singlepoint/dataset_models.py#L30) in QCFractal. They're poorly documented so I'll add some more detail here.

Reply:
```suggestion
        * The input's ``attributes`` attribute (set on QCFractal DatasetEntry objects, such as ``SinglepointDatasetEntry`` and ``TorsiondriveDatasetEntry``)
```